crates/wasmtime/src/runtime/code_memory.rs

```diff
@@ -19,8 +19,12 @@ pub struct CodeMemory {
     // dropped first since it refers to memory owned by `mmap`.
     mmap: ManuallyDrop<MmapVec>,
     unwind_registration: ManuallyDrop<Option<UnwindRegistration>>,
+    #[cfg(feature = "debug-builtins")]
+    debug_registration: ManuallyDrop<Option<crate::runtime::vm::GdbJitImageRegistration>>,
```

Comment:
Reading over this again, mind removing the `ManualllyDrop` wrapper here? This is already paying the cost of the `Option` overhead so I don't think the unsafe drop part is buying a whole lot here.

The destructor can still be early executed through `let _ = self.debug_registration.take();` in the `Drop` implementation though.

(I realize you were probably copying `unwind_registration` and I think we should apply this change there too, as a separate PR if you'd prefer to leave that as-is for now)

Reply:
While we're on the topic - I don't quite get this manual `drop`ping. Wouldn't reordering `mmap` to be after the registrations work equivalently?

Reply:
I think so yeah, but the drop order of fields with respect to field orderings in structs is pretty subtle so we've occasionally used `ManuallyDrop` to make it more clear that the drop order changing is an intentional change. 

Once both `debug_registration` and `unwind_registration` are just `Option` though it would be fine to remove all the `ManuallyDrop` here and the `Drop` implementation would just be `let _ = self.field.take()` for the two registrations.